lib/parallel_tests/cli.rb

```diff
@@ -6,6 +6,8 @@
 module ParallelTests
   class CLI
     def run(argv)
+      Signal.trap("INT") { Thread.new { ParallelTests.stop_all_processes } }
```

Comment:
why the thread here ?

Reply:
if things still hang should give the user some control back ... 1st Ctrl+C=stop 2nd Ctrl+C=kill ... maybe with a counter ...

Reply:
With some versions of Ruby, `mutex.synchronize` can't be called in a trap context. I took Foreman's solution by wrapping it in a thread. https://github.com/maz/foreman/commit/191438115ddc0e94675f4646f105cda3eab7b50b

I'll see if I can work in the counter

Reply:
maybe add that as comment, good to know :)